src/main/java/aquality/appium/mobile/elements/actions/ElementTouchActions.java

```diff
@@ -0,0 +1,114 @@
+                getBottomRightCornerPoint(),
+                getITouchActionsConfiguration().getHorizontalSwipeTopPointXCoefficient(),
+                getITouchActionsConfiguration().getHorizontalSwipeTopPointYCoefficient());
+        this.swipeLeftStartPoint = recalculatePointCoordinates(
+                getBottomRightCornerPoint(),
+                getITouchActionsConfiguration().getVerticalSwipeRightPointXCoefficient(),
+                getITouchActionsConfiguration().getVerticalSwipeRightPointYCoefficient());
+        this.swipeLeftEndPoint = recalculatePointCoordinates(
+                getBottomRightCornerPoint(),
+                getITouchActionsConfiguration().getVerticalSwipeLeftPointXCoefficient(),
+                getITouchActionsConfiguration().getVerticalSwipeLeftPointYCoefficient());
+        this.scrollUpStartPoint = this.scrollDownEndPoint;
+        this.scrollUpEndPoint = this.scrollDownStartPoint;
+        this.swipeRightStartPoint = this.swipeLeftEndPoint;
+        this.swipeRightEndPoint = this.swipeLeftStartPoint;
+    }
+
+    @Override
+    public void swipe(Point endPoint) {
+        AqualityServices.getTouchActions().swipe(element.getElement().getCenter(), endPoint);
+    }
+
+    @Override
+    public void swipeWithLongPress(Point endPoint) {
+        AqualityServices.getTouchActions().swipeWithLongPress(element.getElement().getCenter(), endPoint);
+    }
+
+    @Override
+    public void scrollToElement(SwipeDirection direction) {
+        int numberOfRetries = AqualityServices.get(ITouchActionsConfiguration.class).getSwipeRetries();
+        ITouchActions touchActions = AqualityServices.getTouchActions();
+        while (numberOfRetries > 0) {
+            if (!element.state().isDisplayed()) {
+                switch (direction) {
+                    case DOWN:
+                        touchActions.swipe(scrollDownStartPoint, scrollDownEndPoint);
+                        break;
+                    case UP:
+                        touchActions.swipe(scrollUpStartPoint, scrollUpEndPoint);
+                        break;
+                    case LEFT:
+                        touchActions.swipe(swipeLeftStartPoint, swipeLeftEndPoint);
+                        break;
+                    case RIGHT:
+                        touchActions.swipe(swipeRightStartPoint, swipeRightEndPoint);
+                        break;
+                    default:
+                        throw new IllegalArgumentException(
+                                String.format("'%s' direction does not exist", direction.toString()));
+                }
+            }
+            numberOfRetries--;
+        }
+    }
+
+    /**
+     * Returns Point in the bottom right corner of the screen.
+     */
+    private Point getBottomRightCornerPoint() {
+        Dimension screenSize = AqualityServices.getApplication().getDriver().manage().window().getSize();
+        return new Point(screenSize.width, screenSize.height);
```

Comment:
It won't work everywhere cause. For example, on iOS you have to scroll certaine element on the screen (View), which can be a part of the all screen. So, you have to calculate start and end points relative to the certain elemnt on the screen.